consensus/polybft/state_transaction.go

```diff
@@ -80,6 +80,10 @@ func (cm *CommitmentMessageSigned) VerifyStateSyncProof(proof []types.Hash,
 		return errors.New("no state sync event")
 	}
 
+	if stateSync.ID.Uint64() < cm.Message.StartID.Uint64() {
```

Comment:
`VerifyStateSyncProof` is only used for unit tests in edge client, since proof verification is done in smart contracts, but nevertheless, this is not needed, since if it does not belong to the commitment, `VerifyProof` on merkle tree will return an error (meaning the state sync does not belong to given commitment).
I mean we can leave it as an optimization, but I would say then add the check for the `stateSyncID > commitment.EndID`.